### PR TITLE
Backup database script

### LINKED ISSUE #37: meta: backup instructions
To prevent data loss, the VLA wants to periodically back up the data of our application in an automated way.

We should write instructions on how to back up `vlaorga_postgres_data` docker volume into our setup documentation.

### PR DESCRIPTION
**Pull request by CarolinRoell**
Adds scripts and documentation for backing up and restoring the db.

closes #37

---

### Implementation

- [ ] Does this code change accomplish what it is supposed to do?
- [ ] Can this solution be simplified without unreasonable effort
- [ ] Is the code modular enough?
- [ ] Does similar functionality already exist in the codebase? If yes, why isn’t it reused?
- [ ] Single Responsibility Principle: Does any class serve only one purpose? If not, it should probably split up in multiple classes.
- [ ] Interface Segregation: If interface implemented, is it fully implemented?
- [ ] Dependency Injection: Are central object instantiation processes managed by a single context?
- [ ] If any functionality is missing, is it marked with a comment that starts with TODO:?
- [ ] Do you see any potential to improve the performance of the code significantly? 

### Logic Errors and Bugs

- [ ] Can you think of any use case in which the code does not behave as intended?
- [ ] Can you think of any inputs or external events that could break the code?
- [ ] Are edge cases covered by tests?

### Error Handling and Logging

- [ ] Is error handling done the correct way?
- [ ] Should any logging or debugging information be added or removed?
- [ ] Are error messages user-friendly?

### Dependencies (internal and external)

- [ ] Were updates to documentation, configuration, or readme files made as required by this change?
- [ ] Is the API.md up-to-date?
- [ ] Are the enumerations between frontend and backend synchronized?

### Security and Data Privacy

- [ ] Does the code introduce any security vulnerabilities?
- [ ] Is (user) input validated, sanitized, and escaped to prevent security attacks such as cross-site scripting or SQL injection?
- [ ] Is sensitive data like user data handled and stored securely?
- [ ] Is the right encryption used?
- [ ] Is data retrieved from external APIs or libraries checked for security issues?
- [ ] Were any secrets committed?

### Testing and Testability (later)

- [ ] Have automated tests been added, or have related ones been updated to cover the change?
- [ ] Do the existing tests reasonably cover the code change?
- [ ] Are there some test cases, input or edge cases that should be tested in addition?

### Usability and Accessibility

- [ ] Is the proposed solution (UI) accessible (e.g. colorblindness, ...)?
- [ ] Is the API/UI intuitive to use?
- [ ] Was the app tested on another device?

### Readability

- [ ] Is the code easy to understand?
- [ ] Which parts were confusing to you and why?
- [ ] Can the readability of the code be improved by smaller methods/different names?
- [ ] Is the code located in the right file/folder?
- [ ] Do you think certain methods should be restructured to have a more intuitive control flow?
- [ ] Are there redundant or outdated comments?
- [ ] Could some comments convey the message better?
- [ ] Could some comments be removed by making the code itself more readable?

### CODE REVIEW

File: docs/backupDb.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Backup of VLAOrga Database
+
+## Check environment
+Check that the .env file is correct.
```
Comment (JohaJung) [inline docs/backupDb.md]: This sentence misses context, I would at least add ", i.e. all keys from .env.template are present"

Or leave out this paragraph entirely, since in a running production environment (hopefully?) everything should already be set.

File: docs/backupDb.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+## Create Backup
+For this, the docker of the database has to run.
+
+If there is no backups folder in VLAOrga, create one.
```
[the remark this answers is not in the document]
Comment (JohaJung) [reply docs/backupDb.md]: No, our script should take care of creating that folder (`mkdir -p`)

File: scripts/createBackup.sh [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+source .env
+mkdir -p backups
+docker compose exec -t db pg_dumpall --clean --username=$POSTGRES_USER > backups/vladb_dump_`date +%Y-%m-%d"_"%H_%M_%S`.sql
+echo Dump created
```
Comment (MarcoLeipnitz) [inline scripts/createBackup.sh]: Backup successfully created

File: scripts/restoreBackup.sh [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+fi
+
+source .env
+docker compose --file docker-compose.prod.yml stop prod-app
+docker compose down --volumes db # deletes current db
+docker compose --file docker-compose.prod.yml up db -d
+sleep 5s # wait for docker to boot
+cat $PG_FILE | docker compose exec --no-tty db psql --quiet --no-psqlrc --username=$POSTGRES_USER --dbname=$POSTGRES_DB
+# some errors here are normal, see https://www.postgresql.org/docs/current/app-pg-dumpall.html under notes
+echo Dump restored
```
Comment (MarcoLeipnitz) [inline scripts/restoreBackup.sh]: Dump applied successfully

**Review by MarcoLeipnitz — CHANGES_REQUESTED** [review]
Just a bit more output for the user, otherwise this looks great

File: scripts/restoreBackup.sh [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    echo "Please specify an existing PostgreSQL file."
+    echo "Example: ./applyDbBackup.sh backups/vladb_dump_2026-03-02_15_53_47.sql"
+    exit 1
+fi
+
+source .env
+docker compose --file docker-compose.prod.yml stop prod-app
+docker compose down --volumes db # deletes current db
+docker compose --file docker-compose.prod.yml up db -d
+sleep 5s # wait for docker to boot
```
Comment (JohaJung) [inline scripts/restoreBackup.sh]: Do we have a more intelligent way of recognizing whether the container is started? Maybe some loop trying every second?

Especially, the test server we got from the VLA is slow and may need more than 5 seconds...

**Review by JohaJung — APPROVED** [review]
I won't resist on those two points, but consider them useful and improving the User (or Admin) experience.
